docs/scos/dev/guidelines/security-guidelines.md

```diff
@@ -0,0 +1,150 @@
+
+{% endinfo_block %}
+
+As HTTP is a textual protocol having no built-in encryption, passwords and customer personal data are transferred to shops in plain text. So, a good practice is to configure and implement transport layer security (TLS), which is widely known to most users as HTTPS.
+
+In most cases, it prevents eavesdropping on traffic of users in local public networks like free Wi-Fi hotspots. Besides, it can be used to authenticate users using third-party integrations by requiring a client certificate to be trusted.
+
+TLS configuration is configured on the webserver level. See the following configuration references for both web servers:
+
+* [Nginx](https://nginx.org/en/docs/http/configuring_https_servers.html)
+* [Apache](https://httpd.apache.org/docs/2.4/ssl/ssl_howto.html)
+
+### Forced encrypted communication
+
+Optionally, you can force HTTPS for the Storefront, Back Office, and Glue using the  `Strict-Transport-Security` header:
+* `HttpConstants::ZED_HTTP_STRICT_TRANSPORT_SECURITY_ENABLED`
+* `HttpConstants::YVES_HTTP_STRICT_TRANSPORT_SECURITY_ENABLED`
+* `HttpConstants::ZED_HTTP_STRICT_TRANSPORT_SECURITY_CONFIG`
+* `HttpConstants::YVES_HTTP_STRICT_TRANSPORT_SECURITY_CONFIG`
+* `HttpConstants::GLUE_HTTP_STRICT_TRANSPORT_SECURITY_ENABLED`
+* `HttpConstants::GLUE_HTTP_STRICT_TRANSPORT_SECURITY_CONFIG`
+
+To test web server configuration, use online tools like [SSL Server Test](https://www.ssllabs.com/ssltest/).
+
+## Session security and hijacking
+
+Websites include many third-party JavaScript libraries that can access the content of a page.
+
+* To prevent access to session cookies from Javascript, Spryker sets the HttpOnly attribute of the session cookie by default. We advise to set this attribute for all sensitive cookies.
+* When using TLS, you can use a `secure` cookie flag to instruct a browser to send this cookie back to the server only via an encrypted connection. To configure it, use the  `*_SESSION_COOKIE_SECURE` configuration keys.
+* To prevent session fixation, session identifier is refreshed on login and logout events. We recommend implementing the same behavior for other sensitive cookies if you use them in your shop.
+* Make sure that your web server configuration does not cut these flags from cookie headers.
+* Make sure that `*_SESSION_COOKIE_DOMAIN` matches only your domain to disallow a browser to send the cookie to another domain or subdomain.
+* Never send a session ID as a GET parameter of a URL, because the ID can be logged in logs or forwarded to external websites in HTTP Referer header.
+
+## Cross-site request forgery (CSRF)
+
+CSRF forces a user to execute unwanted actions while being logged in and either clicking on a specially crafted link or just embedding the URL into some HTML tags triggering the request automatically. For example, the `src` attribute of an `img`. To prevent such attacks, Symfony Form provides the `csrf_protection` token by default. We recommend using it in all forms.
```

Comment:
```suggestion
CSRF forces a user to execute unwanted actions while being logged in and either click on a specially crafted link or just embed the URL into some HTML tags triggering the request automatically—for example, the `src` attribute of `img`. To prevent such attacks, Symfony Form provides the `csrf_protection` token by default. We recommend using it in all forms.
```